algorithms/linfa-logistic/src/lib.rs

```diff
@@ -17,115 +17,28 @@
 //!
 
 pub mod error;
+pub mod hyperparams;
 
 use crate::error::{Error, Result};
+use crate::hyperparams::LogisticRegressionValidParams as ValidParams;
+
 use argmin::prelude::*;
 use argmin::solver::linesearch::MoreThuenteLineSearch;
 use argmin::solver::quasinewton::lbfgs::LBFGS;
+
 use linfa::prelude::{AsTargets, DatasetBase};
 use linfa::traits::{Fit, PredictInplace};
 use ndarray::{s, Array, Array1, ArrayBase, Data, Ix1, Ix2};
-use std::default::Default;
 
 mod argmin_param;
 mod float;
 
 use argmin_param::ArgminParam;
 use float::Float;
 
-/// A two-class logistic regression model.
-///
-/// Logistic regression combines linear models with
-/// the sigmoid function `sigm(x) = 1/(1+exp(-x))`
-/// to learn a family of functions that map the feature space to `[0,1]`.
-///
-/// Logistic regression is used in binary classification
-/// by interpreting the predicted value as the probability that the sample
-/// has label `1`. A threshold can be set in the [fitted model](struct.FittedLogisticRegression.html) to decide the minimum
-/// probability needed to classify a sample as `1`, which defaults to `0.5`.
-///
-/// In this implementation any binary set of labels can be used, not necessarily `0` and `1`.
-///
-/// l2 regularization is used by this algorithm and is weighted by parameter `alpha`. Setting `alpha`
-/// close to zero removes regularization and the problem solved minimizes only the
-/// empirical risk. On the other hand, setting `alpha` to a high value increases
-/// the weight of the l2 norm of the linear model coefficients in the cost function.
-///
-/// ## Examples
-///
-/// Here's an example on how to train a logistic regression model on the `winequality` dataset
-/// ```rust
-/// use linfa::traits::{Fit, Predict};
-/// use linfa_logistic::LogisticRegression;
-///
-/// // Example on using binary labels different from 0 and 1
-/// let dataset = linfa_datasets::winequality().map_targets(|x| if *x > 6 { "good" } else { "bad" });
-/// let model = LogisticRegression::default().fit(&dataset).unwrap();
-/// let prediction = model.predict(&dataset);
-/// ```
-pub struct LogisticRegression<F: Float> {
-    alpha: F,
-    fit_intercept: bool,
-    max_iterations: u64,
-    gradient_tolerance: F,
-    initial_params: Option<(Array1<F>, F)>,
-}
-
-impl<F: Float> Default for LogisticRegression<F> {
-    fn default() -> LogisticRegression<F> {
-        LogisticRegression::new()
-    }
-}
-
 type LBFGSType<F> = LBFGS<MoreThuenteLineSearch<ArgminParam<F>, F>, ArgminParam<F>, F>;
 
-impl<F: Float> LogisticRegression<F> {
```

Comment:
LOL this is gonna have a massive merge conflict with my logistic regression changes.

Reply:
if it is the case (which I don't think) that this PR is merged prior to the multinomial logistic one, then I'll exclude the changes to make this PR orthogonal